Recommend the Visual Studio SPA project templates (#30333)
dotnet/AspNetCore.Docs: aspnetcore/includes/spa-templates-recommend-vs.md

---
author: tdykstra
ms.author: tdykstra
ms.date: 09/12/2023
ms.topic: include
---

> Starting with ASP.NET Core 8.0, we recommend the Visual Studio project templates for creating single-page apps (SPAs) based on JavaScript frameworks such as [Angular](https://angular.io/), [React](https://facebook.github.io/react/), and [Vue](https://vuejs.org/). These templates:

Reviewer: ```suggestion
Visual Studio provides project templates for creating single-page apps (SPAs) based on JavaScript frameworks such as [Angular](https://angular.io/), [React](https://facebook.github.io/react/), and [Vue](https://vuejs.org/) that have an ASP.NET Core backend. These templates:
```

* Create a Visual Studio solution with a frontend project and a backend project.
* Use the Visual Studio project type for JavaScript and TypeScript (*.esproj*) for the frontend.

> * Use a [minimal API](xref:fundamentals/minimal-apis/overview) project for the backend.

Reviewer: There are template options for using both an API controller or a minimal API.

```suggestion
* Use an ASP.NET Core project for the backend.
```
